pubtools/pulplib/_impl/client/client.py

```diff
@@ -363,3 +372,8 @@ def _delete_upload_request(self, upload_id):
 
         LOG.debug("Deleting upload request %s", upload_id)
         return self._request_executor.submit(self._do_request, method="DELETE", url=url)
+
+    def _do_get_maintenance(self):
+        url = os.path.join(self._url, "pulp/isos/redhat-maintenance/repos.json")
+
+        return self._request_executor.submit(self._do_request, method="GET", url=url)
```

Comment:
What happens if file is missing (404)?  In my opinion we should be returning some default indicating that the system is not under maintenance.

Reply:
Hm, from JIRA description, seems like there will always be a repos.json file, if some repos in maintenance mode, then there will be entires in repos:{repo_id:{...}}, if not, then just no entires there, but the file is always there. 
I'm thinking this feature isn't that useful?

Reply:
You mean it wouldn't be useful because in practice you always expect the file to be present on the real system?

I think it is in fact still useful to tolerate its absence because it lowers the bar for getting a working environment for use with this library (and tools built on top of it).  For example, imagine that there is a Pub feature built on this library which somehow displays or reacts to maintenance mode, e.g. refuses to run a task if maintenance mode is set, and imagine you are trying to develop with Pub locally to make unrelated changes to that task.

If you say, "it's only valid to use this library against a Pulp server which has the maintenance repo and the repos.json file", you just made life harder for that developer.  Now the developer can't simply run Pub and connect it with some clean local Pulp dev env; they also have to ensure that Pulp server has the repo and the file, or else they immediately hit a wall.

The same applies for autotests - if you require that this file must always be there or this library will crash, then you also forced anyone mocking out the Pulp server at the HTTP level to know about this URL and know how to provide valid data for it (instead of just e.g. defaulting to 404 for unknown URLs, which is reasonable).

So if you make this code only work when this file is present, you just made it harder to set up a working test or development environment.